travis-ci/terraform-config
# GCE NAT rework

## What is the problem that this PR is trying to fix?

The current production (and staging) job VM instances all have public IP addresses, which makes it difficult (impossible?) to monitor or alter their respective network traffic.

## What approach did you choose and why?

(Re)introduce NAT to GCE, mostly, by reviving elder work and doing stuff per [GCE docs](https://cloud.google.com/vpc/docs/special-configurations#multiple-natgateways).

## How can you test this?

Bring up an instance manually with the `no-ip` tag and verify the routing through a NAT host.

<del>
**NOTE**: ⚠️ I believe the current state of this PR is unsafe for use with the `gce-production-.*` projects without destroying stuff that should not be destroyed.
</del>
